BitRepresentation.Deriving fails on tuples

```haskell
module RepresentTuple where

import Clash.Prelude
import Clash.Annotations.BitRepresentation.Deriving

data AB = A (Bool,Bool) | B
deriveDefaultAnnotation [t|AB|]
```
```
RepresentTuple.hs:1:1: error:
    Exception when trying to run compile-time code:
      Unexpected type: TupleT 2
CallStack (from HasCallStack):
  error, called at src/Clash/Annotations/BitRepresentation/Deriving.hs:177:19 in clash-prelude-0.99-inplace:Clash.Annotations.BitRepresentation.Deriving
    Code: deriveDefaultAnnotation [t| AB |]
```

--- a/clash-prelude/src/Clash/Annotations/BitRepresentation/Deriving.hs
+++ b/clash-prelude/src/Clash/Annotations/BitRepresentation/Deriving.hs
@@ -174,6 +174,7 @@ resolve nmap (VarT n) = nmap Map.! n
 resolve nmap (AppT t1 t2) = AppT (resolve nmap t1) (resolve nmap t2)
 resolve _nmap t@(ConT _) = t
 resolve _nmap t@(LitT _) = t
+resolve _nmap t@(TupleT _) = t
 resolve _nmap t = error $ "Unexpected type: " ++ show t
 
 resolveCon :: NameMap -> Con -> Con

--- a/tests/shouldwork/CustomReprs/Deriving/BitPackDerivation.hs
+++ b/tests/shouldwork/CustomReprs/Deriving/BitPackDerivation.hs
@@ -11,9 +11,9 @@ data Train
       SmallInt
       -- ^ Number of wagons
   | Freight
-      SmallInt
+      ( SmallInt
       -- ^ Number of wagons
-      SmallInt
+      , SmallInt )
       -- ^ Max weight
   | Maintenance
   | Toy
@@ -34,7 +34,7 @@ testBench = done'
   where
     testInput = stimuliGenerator $ Toy
                                 :> Maintenance
-                                :> Freight 2 3
+                                :> Freight (2, 3)
                                 :> Passenger 1
                                 :> Nil
 

